### ISSUE TITLE
Dev experience

### ISSUE THREAD

**Issue by Eomm** [fastify/website-next#46]
We must work to improve the dev experience before working on this website takes days to run and build.

The issues to solve are:

- [ ] load a subset of versions to speed up the `npm start` command. Note that it exists the [`onlyIncludeVersions`](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#onlyIncludeVersions) for this purpose
- [ ] generate static contents (`plugins.json`, `benchmarks.json` etc..) on the fly because the website will not start without these mandatories files
- [ ] suggestions?

**Comment by jonnydgreen** [comment]
I'll load up the website locally and report back if I think of any suggestions :)

**Comment by AugustinMauroy** [comment]
Should we use typescript (`tsx`)?

**Comment by AugustinMauroy** [comment]
I can't launch the project locally. It should be enough to do an `npm install` then `npm start`.

**Comment by AugustinMauroy** [comment]
What do you think about creating a slack channel to discuss the project?

**Comment by Eomm** [comment]
> What do you think about creating a slack channel to discuss the project?

we can create a discord one - will do later today

> I can't launch the project locally

Please, read the readme and run `npm run build:website` too. of course, this is one of the parts we need to improve 😄

**Comment by Eomm** [comment]
I have created the discord channel: https://discord.com/channels/725613461949906985/1067900422376935504

**Comment by AugustinMauroy** [comment]
@Eomm  I can't access follow this https://support.discord.com/hc/en-us/articles/204155938-How-do-I-invite-friends-to-my-server-

**Comment by Eomm** [comment]
You can find it in the main fastify repo
https://discord.gg/fastify

**Comment by aradwann** [comment]
- When I opened the repo in Vscode locally, the colors changed, I think it is too vibrant for me 😅
![Screenshot from 2023-03-08 16-40-51](https://user-images.githubusercontent.com/52883485/223743031-8ace25dd-2a6f-4086-a815-f39b109d1049.png)
is it necessary to keep the colors saved in Vscode JSON?

- I wonder if there could another way other than installing GitHub CLI to run the website locally.

**Comment by Eomm** [comment]
> When I opened the repo in Vscode locally, the colors changed, I think it is too vibrant for me 

Here my statement https://github.com/fastify/website-next/pull/60#pullrequestreview-1270560143
TLDR: Yes, remove it and send a PR, but I think the suggested plugins are useful

> I wonder if there could another way other than installing GitHub CLI to run the website locally.

Yes, but we need someone to replace it with bash scripting + we will need **another** GitHub token. the client is handy because it deals with it without annoying the dev IMHO

**Comment by metcoder95** [comment]
Sadly I've never contributed to the website repo, but my take from the comments it seems that we are mostly lacking portability to be able to run the project easily in locally.

Is there anything else do you think is missed or requires re-work?

**Comment by msbn21** [comment]
@Eomm, I am trying to launch the website locally on my machine, I am following the instructions provided in `fastify/website` repo to launch the website locally, however npm start command keeps complaining about missing `static/generated/benchmarks.json` file while running `docusaurus start`, can someone please help me with this ?

**Comment by Eomm** [comment]
What is the output of `npm run build:website`?
What is your system?

**Comment by msbn21** [comment]
I am using mac, here is the output when i run `npm run start:dev`

> @fastify/website@0.0.0 start:dev
> NODE_ENV=development docusaurus start

[INFO] Starting the development server...
[ERROR] Error: Missing generated file: static/generated/benchmarks.json. Run the "npm run build:website" once.
    at /Users/MCX060/Documents/Docs/Docs/Playground/JavaScript/OpenSource/Fastify/Website/fastify_website/docusaurus.config.utils.js:28:13
    at Array.forEach (<anonymous>)
    at Object.checkGeneratedData (/Users/MCX060/Documents/Docs/Docs/Playground/JavaScript/OpenSource/Fastify/Website/fastify_website/docusaurus.config.utils.js:25:18)
    at Object.<anonymous> (/Users/MCX060/Documents/Docs/Docs/Playground/JavaScript/OpenSource/Fastify/Website/fastify_website/docusaurus.config.js:15:3)
    at Module._compile (node:internal/modules/cjs/loader:1233:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at module.exports (/Users/MCX060/Documents/Docs/Docs/Playground/JavaScript/OpenSource/Fastify/Website/fastify_website/node_modules/import-fresh/index.js:32:59)
[INFO] Docusaurus version: 2.4.1
Node version: v20.4.0

**Comment by msbn21** [comment]
then i ran `npm run build:website` but that doesn't seem to generate that file as well. It complains about missing `versioned_docs/version-latest/Guides/Ecosystem.md` file and then fails with the same error

**Comment by msbn21** [comment]
I am using a mac, when i run `npm run build:website` it downloads the releases and then it throws this error
`node:internal/process/promises:289`
            `triggerUncaughtException(err, true /* fromPromise */);`
            `^`

`Error: Client network socket disconnected before secure TLS connection was established`
`    at connResetException (node:internal/errors:720:14)`
`    at TLSSocket.onConnectEnd (node:_tls_wrap:1682:19)`
`    at TLSSocket.emit (node:events:524:35)`
`    at endReadableNT (node:internal/streams/readable:1378:12)`
`    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {`
`  code: 'ECONNRESET',`
`  path: undefined,`
`  host: 'raw.githubusercontent.com',`
`  port: 443,`
`  localAddress: null`
`}`

And then it fails with error `[ERROR] Error: Missing generated file: static/generated/benchmarks.json. Run the "npm run build:website" once.`

**Comment by Eomm** [comment]
It seems an error connecting to github. are you logged in with your cli?
could you add all the output? it help us to understand which step of the process blocks you

**Comment by msbn21** [comment]
Yes, it was github connectivity indeed. I have logged in but the connection was somehow not getting established, i am able to launch the application now. Please let me know if there are any immediate tasks which i can take up and start contributing.

**Comment by Eomm** [comment]
I would start from this one: https://github.com/fastify/website/issues/131

No idea here - but you can add be creative!
